perf/opengrep-scripts/bench.py

```diff
@@ -0,0 +1,184 @@
+#! /usr/bin/env python3
+
+import os
+import subprocess
+import sys
+import time
+import csv
+from urllib.parse import urlparse
+from datetime import datetime
+
+# SETUP
+# (TODO: make some of these cli options?)
+
+repositories = [ ("https://github.com/jellyfin/jellyfin","a0931baa8eb879898f4bc4049176ed3bdb4d80d1")
+               , ("https://github.com/grafana/grafana", "afcb55156260fe1887c4731e6cc4c155cc8281a2")
+               , ("https://gitlab.com/gitlab-org/gitlab", "915627de697e2dd71fe8205853de51ad3794f3ac")
+               , ("https://github.com/Netflix/lemur", "28b9a73a83d350b1c7ab71fdd739d64eec5d06aa")
+               , ("https://github.com/pythongosssss/ComfyUI-Custom-Scripts","943e5cc7526c601600150867a80a02ab008415e7")
+               , ("https://github.com/pmd/pmd", "81739da5caff948dbcd2136c17532b65c726c781")
+               , ("https://github.com/square/leakcanary", "bf5086da26952e3627f18865bb232963e4d019c5")
+               ]
+
+# -j 8 to leave some cores free in case of a random OS heave job in the background
+def regular_cmd(repo):
+    return ["opengrep", "scan", "-c", "rules", repo, "-j", "8"]
```

Comment:
Isn't there a way to get the number of available cpus and subtract 1, would be more robust.

Reply:
Something like: 

```python
import os

num_cpus = min(1, os.cpu_count() - 1)
```

Reply:
Won't it call the python entrypoint if a venv is active though?
Maybe we can add `--experimental` if we want to be deterministic about this?

Reply:
sorry not `min` but `max`